// @umijs/plugin-layout/lib/layout/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _umi = require("umi");

var _pathToRegexp = _interopRequireDefault(require("path-to-regexp"));

var _proLayout = _interopRequireDefault(require("@ant-design/pro-layout"));

require("./style.less");

var _ErrorBoundary = _interopRequireDefault(require("../component/ErrorBoundary"));

var _renderRightContent = _interopRequireDefault(require("./renderRightContent"));

var _Exception = require("../component/Exception");

var _getLayoutConfigFromRoute = _interopRequireDefault(require("../utils/getLayoutConfigFromRoute"));

var _getMenuFromRoute = _interopRequireDefault(require("../utils/getMenuFromRoute"));

var _logo = _interopRequireDefault(require("../assets/logo.svg"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var BasicLayout = function BasicLayout(props) {
  var children = props.children,
      userConfig = props.userConfig,
      location = props.location,
      route = props.route,
      restProps = _objectWithoutProperties(props, ["children", "userConfig", "location", "route"]);

  var _route$routes = route.routes,
      routes = _route$routes === void 0 ? [] : _route$routes;
  var initialInfo = _umi.useModel && (0, _umi.useModel)('@@initialState') || {
    initialState: undefined,
    loading: false,
    setInitialState: null
  }; // plugin-initial-state 未开启

  var initialState = initialInfo.initialState,
      loading = initialInfo.loading,
      setInitialState = initialInfo.setInitialState;

  var _routes = require('@@/core/routes').routes; // 国际化插件并非默认启动


  var intl = _umi.useIntl && (0, _umi.useIntl)();
  var layoutConfig = (0, _getLayoutConfigFromRoute.default)(_routes);

  var patchMenus = userConfig.patchMenus || function (ms) {
    return ms;
  };

  var menus = patchMenus((0, _getMenuFromRoute.default)(routes), initialInfo); // layout 是否渲染相关

  var pathName = location.pathname;
  var layoutRender = {}; // 动态路由匹配

  var currentMatchPaths = Object.keys(layoutConfig).filter(function (item) {
    return (0, _pathToRegexp.default)("".concat(item, "(.*)")).test(pathName);
  });
  var currentPathConfig = currentMatchPaths.length ? layoutConfig[currentMatchPaths[currentMatchPaths.length - 1]] : undefined;

  if (currentPathConfig === null || currentPathConfig === void 0 ? void 0 : currentPathConfig.hideMenu) {
    layoutRender.menuRender = false;
  }

  if (currentPathConfig === null || currentPathConfig === void 0 ? void 0 : currentPathConfig.hideNav) {
    layoutRender.headerRender = false;
  }

  if (currentPathConfig === null || currentPathConfig === void 0 ? void 0 : currentPathConfig.hideLayout) {
    layoutRender.pure = true;
  }

  if (currentPathConfig === null || currentPathConfig === void 0 ? void 0 : currentPathConfig.hideFooter) {
    layoutRender.footerRender = false;
  }

  return /*#__PURE__*/_react.default.createElement(_proLayout.default, _extends({
    route: route,
    location: location,
    title: userConfig.name || userConfig.title,
    className: "umi-plugin-layout-main",
    navTheme: "dark",
    siderWidth: 256,
    onMenuHeaderClick: function onMenuHeaderClick(e) {
      e.stopPropagation();
      e.preventDefault();

      _umi.history.push('/');
    },
    menu: {
      locale: userConfig.locale
    },
    menuDataRender: function menuDataRender() {
      return menus;
    },
    formatMessage: intl && intl.formatMessage,
    logo: _logo.default,
    menuItemRender: function menuItemRender(menuItemProps, defaultDom) {
      if (menuItemProps.isUrl || menuItemProps.children) {
        return defaultDom;
      }

      if (menuItemProps.path) {
        return /*#__PURE__*/_react.default.createElement(_umi.Link, {
          to: menuItemProps.path
        }, defaultDom);
      }

      return defaultDom;
    },
    disableContentMargin: true,
    rightContentRender: function rightContentRender() {
      return (0, _renderRightContent.default)(userConfig, loading, initialState, setInitialState);
    },
    fixSiderbar: true,
    fixedHeader: true
  }, userConfig, restProps, layoutRender), /*#__PURE__*/_react.default.createElement(_ErrorBoundary.default, null, /*#__PURE__*/_react.default.createElement(_Exception.WithExceptionOpChildren, {
    currentPathConfig: currentPathConfig
  }, userConfig.childrenRender ? userConfig.childrenRender(children) : children)));
};

var _default = BasicLayout;
exports.default = _default;
