// @umijs/plugin-layout/lib/utils/getLayoutConfigFromRoute.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

function _react() {
  const data = _interopRequireDefault(require("react"));

  _react = function _react() {
    return data;
  };

  return data;
}

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

const defaultLayoutConfig = {
  hideMenu: false,
  hideNav: false,
  hideFooter: false
};
/*
 * 计算当前路由的最终 layout 配置
 * - 继承父路由的 layout 配置
 * - 但子路由的 layout 配置优先级更高
 */

function calcExecLayoutConfig(parentRouteLayoutConfig, layout) {
  const execConfig = {};

  for (let configItem in parentRouteLayoutConfig) {
    let execConfigItem = parentRouteLayoutConfig[configItem];

    switch (layout) {
      case undefined:
        execConfig[configItem] = execConfigItem;
        break;

      case true:
        execConfig[configItem] = false;
        break;

      case false:
        execConfig[configItem] = true;
        break;

      default:
        execConfig[configItem] = layout[configItem] === undefined ? execConfigItem : layout[configItem];
    }
  }

  return execConfig;
}
/**
 * @param routes
 */


function formatter(routes = [], prefix = '', base = '/', parentRouteLayoutConfig = defaultLayoutConfig, LayoutConfig = {}) {
  routes.filter(item => {
    var _item$path;

    return item && !((_item$path = item.path) === null || _item$path === void 0 ? void 0 : _item$path.startsWith('http'));
  }).map(route => {
    const layout = route.layout,
          indexRoute = route.indexRoute,
          _route$path = route.path,
          path = _route$path === void 0 ? '' : _route$path,
          routes = route.routes,
          unaccessible = route.unaccessible;
    const execLayoutConfig = calcExecLayoutConfig(parentRouteLayoutConfig, layout); // 拼接 path

    const absolutePath = path.startsWith('/') ? path : `${base}${base === '/' ? '' : '/'}${path}`;
    LayoutConfig[`${prefix}${absolutePath}`] = _objectSpread({}, execLayoutConfig, {
      unAccessible: unaccessible || false
    }); // 拼接 childrenRoutes, 处理存在 indexRoute 时的逻辑

    const childrenRoutes = indexRoute ? [_objectSpread({
      path,
      layout
    }, indexRoute)].concat(routes || []) : routes; // 拼接返回的 layout 数据

    if (childrenRoutes && childrenRoutes.length) {
      const result = formatter(childrenRoutes, prefix, absolutePath, execLayoutConfig, LayoutConfig);
      LayoutConfig = _objectSpread({}, LayoutConfig, {}, result);
    }
  });
  return LayoutConfig;
}

var _default = formatter;
exports.default = _default;
