// @ant-design/pro-layout/es/SettingDrawer/BlockCheckbox.js
import "antd/es/tooltip/style";
import _Tooltip from "antd/es/tooltip";
import { CheckOutlined } from '@ant-design/icons';
import React from 'react';
import { getFormatMessage } from './index';
var baseClassName = 'ant-pro-setting-drawer-block-checbox';

var BlockCheckbox = function BlockCheckbox(_ref) {
  var value = _ref.value,
      onChange = _ref.onChange,
      propsList = _ref.list;
  var formatMessage = getFormatMessage();
  var list = propsList || [{
    key: 'sidemenu',
    url: 'https://gw.alipayobjects.com/zos/antfincdn/XwFOFbLkSM/LCkqqYNmvBEbokSDscrm.svg',
    title: formatMessage({
      id: 'app.setting.sidemenu'
    })
  }, {
    key: 'topmenu',
    url: 'https://gw.alipayobjects.com/zos/antfincdn/URETY8%24STp/KDNDBbriJhLwuqMoxcAr.svg',
    title: formatMessage({
      id: 'app.setting.topmenu'
    })
  }];
  return React.createElement("div", {
    className: baseClassName,
    key: value
  }, list.map(function (item) {
    return React.createElement(_Tooltip, {
      title: item.title,
      key: item.key
    }, React.createElement("div", {
      className: "".concat(baseClassName, "-item"),
      onClick: function onClick() {
        return onChange(item.key);
      }
    }, React.createElement("img", {
      src: item.url,
      alt: item.key
    }), React.createElement("div", {
      className: "".concat(baseClassName, "-selectIcon"),
      style: {
        display: value === item.key ? 'block' : 'none'
      }
    }, React.createElement(CheckOutlined, null))));
  }));
};

export default BlockCheckbox;
