// @umijs/preset-built-in/lib/plugins/features/forkTSChecker.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

function _react() {
  const data = _interopRequireDefault(require("react"));

  _react = function _react() {
    return data;
  };

  return data;
}

function _umi() {
  const data = require("umi");

  _umi = function _umi() {
    return data;
  };

  return data;
}

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = api => {
  const deepmerge = _umi().utils.deepmerge;

  api.describe({
    key: 'forkTSChecker',
    config: {
      schema(joi) {
        return joi.object({
          tsconfig: joi.string().description('Path to tsconfig.json file'),
          compilerOptions: joi.object().description('Allows overriding TypeScript options. Should be specified in the same format as you would do for the compilerOptions property in tsconfig.json.'),
          eslint: joi.boolean(),
          eslintOptions: joi.object(),
          async: joi.boolean(),
          ignoreDiagnostics: joi.array().items(joi.number()),
          formatter: joi.valid('default', 'codeframe'),
          formatterOptions: joi.object(),
          silent: joi.boolean(),
          checkSyntacticErrors: joi.boolean()
        }).description('More options see https://www.npmjs.com/package/fork-ts-checker-webpack-plugin#options');
      }

    },
    enableBy: () => {
      var _api$config;

      return process.env.FORK_TS_CHECKER || !!((_api$config = api.config) === null || _api$config === void 0 ? void 0 : _api$config.forkTSChecker);
    }
  });
  api.chainWebpack(webpackConfig => {
    var _api$config2;

    webpackConfig.plugin('fork-ts-checker').use(require('fork-ts-checker-webpack-plugin'), [deepmerge({
      formatter: 'codeframe',
      // parallel
      async: false,
      checkSyntacticErrors: true
    }, ((_api$config2 = api.config) === null || _api$config2 === void 0 ? void 0 : _api$config2.forkTSChecker) || {})]);
    return webpackConfig;
  });
};

exports.default = _default;
